### PR TITLE
[feat + style] Sign Up + Sign In page: basic implementation w/ Supabase and styling

### PR DESCRIPTION
**Pull request by RohinJ444**
[//]: # "Feel free to customize this template to your project's vibes"

### :croissant: Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

Took the basic back-end login and signup functionality I had in one login page earlier and created two new styled pages: a sign-up page (app/auth/signup/page.tsx) and a sign in page (app/auth/signin/page.tsx). I created styled components in app/auth/auth-styles.ts for both pages. The page remains centered with varying screen sizes. I saw some minor style/design changes made to Figma this afternoon and those are reflected in the implementation. 

Confirmation emails need to be configured in Supabase settings, but after that, basic sign-up and sign-in functionality should be fully implemented. I describe some more features that would complete / enhance the authentication process in next steps. For now, I also added an error message at the bottom of the white card in the center of the page to make it easier to validate whether logins/signups were successful.

### :baguette_bread: Screenshots

Screenshot 2024-10-25 at 11 51 04 PM

Screenshot 2024-10-25 at 11 43 04 PM

[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 

## :sandwich: How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

#### Functionality

Basically everything is in the first commit with user authentication and styling. The signup page (app/auth/signup/page.tsx) takes as input an email, password, and confirmed password. It should display an error if passwords do not match or if the email is invalid. The login page (app/auth/signin/page.tsx) takes as input an email and password, and it should display an error if the login credentials are invalid. The supabase user table to track users and auth requests is linked here: https://supabase.com/dashboard/project/paifoskjfzbjbiehraqj/auth/users

#### Styling

Uses flexbox so it should remain centered with varying screen sizes. I believe all fonts, colors, and font sizes should match the Figma now, except for the temporary error message I added as discussed above. All the styled components I created for the page can be found in app/auth/auth-styles.ts

#### Testing

Because the SMTP confirmation email isn't set up yet, sign up functionality isn't quite there right now, but you can test login with the following email + password I added to the users table

Email: rohin.juneja@berkeley.edu
Password: vignesh

## :pie: Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

#### Functionality

- Confirmation emails do not fully work yet—this has to do with Supabase Auth settings (I currently don't have permission to change these). As a result, any users who don't have a @berkeley.edu email will get an error that confirmation email failed when attempting to sign up.
- "Continue with Google" is a dummy button as of now, it does not provide any functionality 
- Does not yet have a "forgot password" option
- Does not yet check whether user with specified email already exists during sign-up 
- Does not redirect anywhere after signup or signin. For now, I added a temporary message state that displays whether sign-ups and logins were successful so I could validate that supabase was actually working. 

#### Styling

- Small thing, but I couldn't get the drop shadow to properly render around the white card idk why :sob:
- Also small, but I don't know if I used the correct shade of blue for the hyperlinks at the bottom of each page

### :pretzel: Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

https://styled-components.com/docs/basics#adapting-based-on-props

Referenced these PRs:
https://github.com/calblueprint/immigration-justice-project/pull/24/files
https://github.com/calblueprint/immigration-justice-project/pull/27/files

### :bagel: Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @celinechoiii

### CODE REVIEW

File: app/auth/auth-styles.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
+import styled from 'styled-components';
+import COLORS from '../../styles/colors';
```
Comment (celinechoiii) [inline app/auth/auth-styles.ts]: update all imports using the alias! for example: `import COLORS from '@/styles/colors';`

File: app/auth/signup/page.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+          <Separator>
+            <span>or</span>
+          </Separator>
+          <GoogleButton>Continue with Google</GoogleButton>
+          {message && <LoginMessage isError={isError}>{message}</LoginMessage>}
+        </Form>
+      </Card>
+      <SmallBuffer />
+      <Footer>
+        Already have an account? <Link href="/auth/signin">Sign in!</Link>
```
Comment (celinechoiii) [inline app/auth/signup/page.tsx]: Update to `<Link href="/signin">`

File: app/auth/signin/page.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+          <Separator>
+            <span>or</span>
+          </Separator>
+          <GoogleButton>Continue with Google</GoogleButton>
+          {message && <LoginMessage isError={isError}>{message}</LoginMessage>}
+        </Form>
+      </Card>
+      <SmallBuffer />
+      <Footer>
+        Don’t have an account? <Link href="/auth/signup">Sign up!</Link>
```
Comment (celinechoiii) [inline app/auth/signin/page.tsx]: Update to `<Link href="/signup">`

**Review by celinechoiii — COMMENTED** [review]
can you change the auth folder name to `(auth)`?
